### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by Piercel2022**
In this project we preserve the data from your app and load preserved data on startup.
On exit from your app save all the data (books, people, and rentals) into JSON files after creating them in the folder data.
On startup of the app,  all data from files are loaded from the files.

### CODE REVIEW

**Review by cynthiainga — COMMENTED** [review]
Hi @Piercel2022 & @looh16, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't use the correct Gitflow, the Pull Request should be made from the “feature branch” into the dev branch. Please kindly fix that 🙃 😉

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: main.rb [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+require './data/data_write'
 
 class Main
   def run
     @app = App.new
+    @persons = read_person
+    @books = read_book
+    @rentals = read_rentals
     options
   end
```
Comment (1GORDON) [inline main.rb:12]: Great job @Piercel2022 🤝

- [ ] [Optional] Kindly make sure there are no errors when you try to add the `book_author`.
error1

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @Piercel2022 ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (looh16) [reply main.rb:12]: Thanks

**Pull request merged**